olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -219,63 +233,67 @@ class AuthenticationClient::Impl final {
       const AuthenticationClient::SignInUserCallback& callback);
 
  private:
-  std::string server_url_;
-  std::shared_ptr<
-      olp::thread::Atomic<utils::LruCache<std::string, SignInResult> > >
-      client_token_cache_;
-  std::shared_ptr<
-      olp::thread::Atomic<utils::LruCache<std::string, SignInUserResult> > >
-      user_token_cache_;
+  std::shared_ptr<SignInCacheType> client_token_cache_;
+  std::shared_ptr<SignInUserCacheType> user_token_cache_;
   http::NetworkSettings network_settings_;
```

Comment:
Do we need this one at all?
network_settings_ used only to setup NetworkProxySettings for requests. AuthenticationSettings covers this data too.

Reply:
It seems that way. You should be able to create that on the fly for each request.